relentless/optimize/method.py

```diff
@@ -145,15 +153,33 @@ class LineSearch:
 
         \mathbf{d} = \mathbf{x}_{end} - \mathbf{x}_{start}
 
-        \hat{\mathbf{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+        \mathbf{\hat{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+
+    :math:`\mathbf{X}=\left[X_1,\ldots,X_n\right]` is defined as the scaling
+    parameters for the gradient such that :math:`y_i=\frac{x_i}{X_i}`. By
+    default, the variables are left unscaled (:math:`X_i=1`).
+
+    With scaling, the gradient of the function becomes:
+
+    .. math::
+
+        \nabla f\left(\mathbf{y}\right) = \left[{X_1}^2 \frac{\partial f}{\partial x_1},
+                                                \cdots,
+                                                {X_n}^2 \frac{\partial f}{\partial x_n}\right]
 
-    Then, a "target" value :math:`t` is defined as:
+    Then, a scalar "target" value :math:`t` is defined as:
 
     .. math::
 
-        t = -\hat{\mathbf{d}} \cdot \nabla{f\left(\mathbf{x}\right)}
+        t = -\mathbf{\hat{d}} \cdot \nabla{f\left(\mathbf{y}\right)}
+
+    which is equivalent to:
+
+    .. math::
 
-    Because :math:`\hat{\mathbf{d}}` is a descent direction, the target at the
+        t = \sum_{i=1}^{n} \left(-\hat{d}_i {X_i}^2 \frac{\partial f}{\partial x_i}\right)
```

Comment:
I see this `^2` is coming up here and it looks like in the code. @zeesherman please double check.

I think we need to be careful about what we are defining as the direction `d` and the target `t`. `d` would pick up an extra factor of X to convert between x and y, but I am not following this clearly from the docs.